### PR TITLE
Add type hints for ViLT models

### PR DESCRIPTION
**Pull request by donelianc**
# What does this PR do?

Adding type hints for ` ViLT` model (PyTorch). Issue #16059.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests), Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? _Task requested [here](https://github.com/huggingface/transformers/issues/16059#issuecomment-1210179575)._
- [x] Did you make sure to update the documentation with your changes? Here are the [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests? _Ran `make fixup` before last commit._

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by donelianc** [comment]
Hi @Rocketknight1, this PR is ready to review. Can you help me here, please?

File: src/transformers/models/vilt/modeling_vilt.py [hunk 1]
```diff
@@ -1436,18 +1436,18 @@ def __init__(self, config):
...
+        pixel_values: Optional[torch.FloatTensor] = None,
+        pixel_mask: Optional[torch.LongTensor] = None,
+        head_mask: Optional[torch.FloatTensor] = None,
+        inputs_embeds: Optional[torch.FloatTensor] = None,
+        image_embeds: Optional[torch.FloatTensor] = None,
+        labels: Optional[torch.LongTensor] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
     ):
```
Comment (Rocketknight1) [inline src/transformers/models/vilt/modeling_vilt.py:1451]: I think the return type is missing here!

**Review by Rocketknight1 — COMMENTED** [review]
This looks really good, thank you! I think there's one missing return type, but other than that this PR is ready to go. If you add the missing return type this should be ready to merge!

File: src/transformers/models/vilt/modeling_vilt.py (hunk 1 above)
Comment (donelianc) [reply src/transformers/models/vilt/modeling_vilt.py]: You are right! I missed that return type. I added it in other commit :)

**Review by Rocketknight1 — APPROVED** [review]
Looks perfect, thank you!

**Pull request merged**